### PR TITLE
API-167 : Change to create real docker image to use in doc generation.

### PR DESCRIPTION
**Pull request by adrienzieba**
https://appdirect.jira.com/browse/API-167

Change to create real docker image to use in doc generation.
Add jenkins pipeline to release docker image
Remove slate-extract ad slate-launch docker image. We just have 1 docker image and we need to specify the goal (server or build) to start correctly.

### CODE REVIEW

**Comment by adrienzieba** [comment]
@patvong @jlepage-appdirect Please review

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+pipeline {
+	agent any
+	stages {
+		stage('Checkout project...') {
+            steps {
+                checkout scm
+            }
+        }
+
+        stage('Building PR') {
```
Comment (jlepage-appdirect) [inline Jenkinsfile]: by curiosity, what were the Jenkinsfile you based this on?
I did a small sampling and there doesn't seem to be "standard" way to define various build stages.

In this case I would believe this is pretty basic release mgt stuff, so if possible would like to stick as close as possible to the mainstream pattern used (or guidelines defined by platform, if that exist).

Comment (adrienzieba) [reply Jenkinsfile]: For example : 
https://github.com/AppDirect/grafana-config/blob/master/Jenkinsfile
https://github.com/AppDirect/pubsub/blob/master/Jenkinsfile

File: slate-extract.sh [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
+if [ ! -z "$1" ];
+then
+	SLATE_VERSION=$1
+fi
+
+echo "Remove docker container named 'slate-extract' if present"
 docker rm --force slate-extract || true
-docker create --name slate-extract slate-extract:build
+echo "Run slate:$SLATE_VERSION to extract sources"
+docker run --env goal_env=build --name slate-extract docker.appdirect.tools/documentation/slate:$SLATE_VERSION
```
Comment (patvong) [inline slate-extract.sh]: This builds the slate HTML without our swagger files right?

Comment (adrienzieba) [reply slate-extract.sh]: right

**Pull request merged**